great_expectations/data_context/store/tuple_store_backend.py

```diff
@@ -254,6 +255,10 @@ def __init__(  # noqa: PLR0913
         base_public_path=None,
         store_name=None,
     ) -> None:
+        if filepath_suffix and filepath_template:
+            warnings.warn("filepath_suffix is ignored when filepath_template is set")
+            filepath_suffix = None
```

Comment:
We have too much functionality on this class but in short, if you pass a template and a suffix, you hit an error.

Making the least changes possible, this will ensure that we get `.json` extensions on our persisted objects by default